src/overlays/actors/ovl_Bg_Spot15_Rrbox/z_bg_spot15_rrbox.c

```diff
@@ -21,45 +39,343 @@ const ActorInit Bg_Spot15_Rrbox_InitVars = {
+    this->dyna.actor.velocity.y = 0.0f;
+    this->dyna.actor.velocity.z = 0.0f;
+}
+
+void func_808B40AC(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->unk_168 <= 0 && fabsf(this->dyna.unk_150) > 0.001f) {
+        if (func_808B3AAC(this, globalCtx) != 0 && func_808B4010(this, globalCtx) == 0) {
+            this->unk_17C = this->dyna.unk_150;
+            func_808B4178(this, globalCtx);
+            return;
+        }
+
+        player->stateFlags2 &= ~0x10;
+        this->dyna.unk_150 = 0.0f;
+        return;
+    }
+    player->stateFlags2 &= ~0x10;
+    this->dyna.unk_150 = 0.0f;
+}
+
+void func_808B4178(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    this->actionFunc = func_808B4194;
+    this->dyna.actor.gravity = 0.0f;
+}
+
+void func_808B4194(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    f32 sign;
+    Player* player = PLAYER;
+    f32 temp_f0_2;
+    s32 approxFResult;
+
+    Actor* actor = &this->dyna.actor;
+
+    this->unk_174 = this->unk_174 + 0.5f;
+    if (2.0f < this->unk_174) {
+        this->unk_174 = 2.0f;
+    } else {
+        this->unk_174 = this->unk_174;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B3A34.s")
+    approxFResult = Math_ApproxF(&this->unk_178, 20.0f, this->unk_174);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B3A40.s")
+    sign = 0.0f <= this->unk_17C ? 1.0f : -1.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B3AAC.s")
+    temp_f0_2 = ((float)sign) * this->unk_178;
+    actor->posRot.pos.x = actor->initPosRot.pos.x + (temp_f0_2 * this->unk_16C);
+    actor->posRot.pos.z = actor->initPosRot.pos.z + (temp_f0_2 * this->unk_170);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/BgSpot15Rrbox_Init.s")
+    if (func_808B3F58(this, globalCtx) == 0) {
+        actor->initPosRot.pos.x = actor->posRot.pos.x;
+        actor->initPosRot.pos.z = actor->posRot.pos.z;
+        player->stateFlags2 &= ~0x10;
+        this->dyna.unk_150 = 0.0f;
+        this->unk_178 = 0.0f;
+        this->unk_174 = 0.0f;
+        func_808B4380(this, globalCtx);
+    } else {
+        if (approxFResult) {
+            player = PLAYER;
+            if (func_808B4010(this, globalCtx) != 0) {
+                Audio_PlayActorSound2(actor, NA_SE_EV_WOOD_BOUND);
+            }
+            if (func_808B3A40(this, globalCtx)) {
+                func_80078884(NA_SE_SY_CORRECT_CHIME);
+            }
+            actor->initPosRot.pos.x = actor->posRot.pos.x;
+            actor->initPosRot.pos.z = actor->posRot.pos.z;
+            player->stateFlags2 &= ~0x10;
+            this->dyna.unk_150 = 0.0f;
+            this->unk_178 = 0.0f;
+            this->unk_174 = 0.0f;
+            this->unk_168 = 10;
+            func_808B4084(this, globalCtx);
+        }
+    }
+    Audio_PlayActorSound2(actor, 0x200AU);
```

Comment:
you can replace `0x200A` with `NA_SE_EV_ROCK_SLIDE - SFX_FLAG`
`SFX_FLAG` has to do with looping the sound, not very well known

Reply:
Fixed.